compiler/rustc_mir_transform/src/check_alignment.rs

```diff
@@ -237,11 +236,10 @@ fn insert_alignment_check<'tcx>(
                 required: Operand::Copy(alignment),
                 found: Operand::Copy(addr),
             }),
-            unwind: if tcx.sess.panic_strategy() == PanicStrategy::Unwind {
-                UnwindAction::Terminate
-            } else {
-                UnwindAction::Unreachable
-            },
+            // The panic symbol that this calls is #[rustc_nounwind]. We never want to insert an
```

Comment:
Would be good to mention `panic_misaligned_pointer_dereference` here so one can grep for what that symbol is. (I was not aware/forgot that different `AssertKind` dispatch to completely different code on assertion failure.)